servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -1736,76 +1567,85 @@ void RendererSceneRenderRD::_process_ssil(RID p_render_buffers, RID p_environmen
 }
 
-void RendererSceneRenderRD::_copy_framebuffer_to_ssil(RID p_render_buffers) {
-	RenderBuffers *rb = render_buffers_owner.get_or_null(p_render_buffers);
-	ERR_FAIL_COND(!rb);
+void RendererSceneRenderRD::_copy_framebuffer_to_ssil(Ref<RenderSceneBuffersRD> p_render_buffers) {
+	ERR_FAIL_COND(p_render_buffers.is_null());
 
-	if (rb->ss_effects.ssil.last_frame.is_valid()) {
-		copy_effects->copy_to_rect(rb->texture, rb->ss_effects.ssil.last_frame, Rect2i(0, 0, rb->width, rb->height));
+	if (p_render_buffers->ss_effects.ssil.last_frame.is_valid()) {
+		Size2i target_size = p_render_buffers->get_target_size();
+		RID texture = p_render_buffers->get_final_texture();
+		copy_effects->copy_to_rect(texture, p_render_buffers->ss_effects.ssil.last_frame, Rect2i(0, 0, target_size.x, target_size.y));
 
-		int width = rb->width;
-		int height = rb->height;
-		for (int i = 0; i < rb->ss_effects.ssil.last_frame_slices.size() - 1; i++) {
+		int width = target_size.x;
+		int height = target_size.y;
+		for (int i = 0; i < p_render_buffers->ss_effects.ssil.last_frame_slices.size() - 1; i++) {
 			width = MAX(1, width >> 1);
 			height = MAX(1, height >> 1);
-			copy_effects->make_mipmap(rb->ss_effects.ssil.last_frame_slices[i], rb->ss_effects.ssil.last_frame_slices[i + 1], Size2i(width, height));
+			copy_effects->make_mipmap(p_render_buffers->ss_effects.ssil.last_frame_slices[i], p_render_buffers->ss_effects.ssil.last_frame_slices[i + 1], Size2i(width, height));
 		}
 	}
 }
 
-void RendererSceneRenderRD::_process_taa(RID p_render_buffers, RID p_velocity_buffer, float p_z_near, float p_z_far) {
-	RenderBuffers *rb = render_buffers_owner.get_or_null(p_render_buffers);
-	ERR_FAIL_COND(!rb);
+void RendererSceneRenderRD::_process_taa(Ref<RenderSceneBuffersRD> p_render_buffers, RID p_velocity_buffer, float p_z_near, float p_z_far) {
+	uint32_t view_count = p_render_buffers->get_view_count();
+	Size2i internal_size = p_render_buffers->get_internal_size();
+	Size2i target_size = p_render_buffers->get_target_size();
 
 	bool just_allocated = false;
-	if (rb->taa.history.is_null()) {
-		RD::TextureFormat tf;
-		if (rb->view_count > 1) {
-			tf.texture_type = RD::TEXTURE_TYPE_2D_ARRAY;
-		}
-		tf.format = _render_buffers_get_color_format();
-		tf.width = rb->internal_width;
-		tf.height = rb->internal_height;
-		tf.array_layers = rb->view_count; // create a layer for every view
-		tf.usage_bits = RD::TEXTURE_USAGE_SAMPLING_BIT | (_render_buffers_can_be_storage() ? RD::TEXTURE_USAGE_STORAGE_BIT : 0);
+	if (!p_render_buffers->has_texture(SNAME("taa"), SNAME("taa_history"))) {
+		uint32_t usage_bits = RD::TEXTURE_USAGE_SAMPLING_BIT | (_render_buffers_can_be_storage() ? RD::TEXTURE_USAGE_STORAGE_BIT : 0);
+
+		RD::DataFormat format = _render_buffers_get_color_format();
+		p_render_buffers->create_texture(SNAME("taa"), SNAME("taa_history"), format, usage_bits);
+		p_render_buffers->create_texture(SNAME("taa"), SNAME("taa_temp"), format, usage_bits);
```

Comment:
Stylistic nitpick, but we could remove `taa_` bit of the texture names, since it's already accessed by the `"taa"` prefix.

Reply:
Will do!